Automattic/newspack-custom-content-migrator
# Feat/guest author validation refactor

## How to test

- Grab a DB export from LSV from or before December.
- Run `wp newspack-content-migrator la-silla-vacia-validate-guest-authors`
- You can kill the script after a few dozen validation errors.
- Looking at the data that is output, find a standalone Guest Author (i.e. one that has an empty `cap-linked_account`, or simply one GA that doesn't show any WP_User data in the output)
- Using the `wp_posts.id`/Guest Author ID and `wp_terms.term_id`/Author Taxonomy ID, run `wp newspack-content-migrator co-authors-set-standalone-guest-author-data --guest-author-id=<guest-author-id> --term-id=<term-id>`. Notice the output, and what data has been changed/updated.

Another way to test this is:
- Create a GA on a test site.
- Add or remove `cap-*` related data from `wp_postmeta` for that GA
- Modify the GA's `wp_terms`/`wp_term_taxonomy` record by updating the slug or description, etc.
- Can you assign that GA to a post? What happens when you publish that post? Is a new Term created?
- After running `wp newspack-content-migrator co-authors-set-standalone-guest-author-data --guest-author-id=<guest-author-id> --term-id=<term-id>`, does the same thing happen when you assign that GA to a post?

---
